readthedocs/core/views/serve.py

```diff
@@ -110,6 +110,14 @@ def inner_view(  # noqa
 def redirect_project_slug(request, project, subproject):  # pylint: disable=unused-argument
     """Handle / -> /en/latest/ directs on subdomains."""
     urlparse_result = urlparse(request.get_full_path())
+
+    # When accessing docs.customdomain.org/projects/subproject/ and the
+    # ``subproject`` is a single-version, we don't have to redirect but to serve
+    # the index file instead.
+    if subproject and subproject.single_version:
+        # TODO: find a way to import ``serve_docs`` from corporate
+        return serve_docs(request, project, project.slug, subproject, subproject.slug)
```

Comment:
Since this is only needed for corporate, the only missing thing in this hacky PR is to import `serve_docs` from corporate before merging (`readthedocsinc.core.views.serve_docs`).